src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2910,7 +2919,35 @@ public GrobidPreferences getGrobidPreferences() {
         return grobidPreferences;
     }
 
-@Override
+    @Override
+    public JournalInformationPreferences getJournalInformationPreferences() {
+        if (Objects.nonNull(journalInfoPreferences)) {
+            return journalInfoPreferences;
+        }
+
+        journalInfoPreferences = new JournalInformationPreferences(getJournalInfoEnablementStatus());
+
+        EasyBind.listen(journalInfoPreferences.enablementStatusProperty(), (obs, oldValue, newValue) -> {
+            putBoolean(JOURNALINFO_ENABLED, newValue == EnablementStatus.ENABLED);
+            putBoolean(JOURNALINFO_DISABLED, newValue == EnablementStatus.DISABLED);
+        });
+
+        return journalInfoPreferences;
+    }
+
+    private EnablementStatus getJournalInfoEnablementStatus() {
+        EnablementStatus enablementStatus = EnablementStatus.FIRST_START;
+
+        if (getBoolean(JOURNALINFO_ENABLED)) {
```

Comment:
Ohoh, two Booleans for the same thing. -- While suggesting the tri-state, I thought, there was an easy check for key presence.

The "hacky" way would be following:

```
if (getSting(JOURNALINFO_ENABLED, "notAvailable").equals("notAvailable") {
    return EnablementStatus.FIRST_START;
} else if (getBoolean(JOURNALINFO_ENABLED)) {
  return EnablementStatus.ENABLED;
} else {
  return EnablementStatus.DISABLED;
}
```

The "clean" way could be to use the string serialization of the Enum in a String.

```` java
public enum EnablementStatus {
    FIRST_START, // The first time a user uses this service
    ENABLED,
    DISABLED;

    public static EnablementStatus fromString(String status) {
        for (EnablementStatus value : EnablementStatus.values()) {
            if (value.toString().equalsIgnoreCase(status)) {
                return value;
            }
        }
        throw new IllegalArgumentException("No enum found with value: " + status);
    }
}
```

With this way, you get rid of `JOURNALINFO_DISABLED`.

Reply:
Yes please. We also use it in other places. Go the extra mile, do it clean, do it right. :-D

Reply:
I was wondering about how the redundant boolean could be removed! will use this approach